### DISCUSSION TITLE
뷰포트 높이에 따른 페이지 사이즈 조정에 대한 논의가 필요합니다.

### DISCUSSION THREAD

**Discussion by doggopawer** [team-nabi/nabi-market-client#48] [Announcements]
**문제 상황**
현재 무한스크롤 관련해서 아래와 같은 이슈가 존재합니다.

뷰포트 높이가 1355px이면서, PAGE_SIZE가 5일때
![스크린샷 2023-11-07 오후 3 43 01](https://github.com/team-nabi/nabi-market-client/assets/107387817/2e0cca83-a332-492d-9051-b249eb746c5f)
=> 무한스크롤 동작 X

뷰포트 높이가 780px이면서, PAGE_SIZE가 5일때
![스크린샷 2023-11-07 오후 3 43 18](https://github.com/team-nabi/nabi-market-client/assets/107387817/adf5acbf-d460-438b-afdd-c1dc8124235c)

=> 무한스크롤 동작 O

**예상원인**
해당 문제가 발생하는 예상 원인은 아마 intersectionObserver의 동작때문인 것 같습니다.
현재 코드에서 entry.isIntersecting 이 변화 할때마다 다음페이지의 데이터를 가져오는 fetchNextPage가 호출이되는데
처음가져오는 데이터의 양이 적기 때문에, isIntersecting이 true로 고정이 되어있어 무한스크롤이 동작하지 않습니다.

**결론 및 요청**
그래서 이부분에 대해 어떻게 논의할 것인지, 혹은 코드적으로 해결책을 가지고 계신분은 알려주시기 바랍니다.

**Comment by oaoong** [comment]
옵저버의 범위와는 별개로 처음 가져오는 데이터 양이 더 늘 필요가 있지 않을까요?
첫 페이지를 채울 정도는 되어야 할 것 같기도 하네요
